tests/app/test_application.py

```diff
@@ -47,3 +50,20 @@ def test_elastic_search_client_performs_retries_on_connection_error(self, perfor
 
         assert perform_request.call_count == 1 + _app_ctx_stack.top.elasticsearch.transport.max_retries
         assert perform_request.call_count == 1 + 3
+
+
+class TestGzip(BaseApplicationTestWithIndex):
+    def setup(self):
+        super().setup()
+        for c in "abc":
+            # create a bunch of indexes with quite long names so we definitely have a response from the "/" route
+            # that is > the minimum gzipped size limit
+            self.create_index(index_name=f"test-{c*128}")
+
+    def test_gzip(self):
+        response = self.client.get('/', headers={"Accept-Encoding": "gzip"})
```

Comment:
Are we going to have to add this kwarg to all our search API client calls in the apps?

Reply:
I don't _think_ so - I think `requests` adds it by default, but 'twould be good to check.